tooling/nargo_cli/src/cli/mod.rs

```diff
@@ -115,14 +107,28 @@ enum NargoCommand {
     GenerateCompletionScript(generate_completion_script_cmd::GenerateCompletionScriptCommand),
 }
 
+/// Commands that can execute on the workspace level, or be limited to a selected package.
+trait WorkspaceCommand {
+    fn package_selection(&self) -> PackageSelection;
+    fn lock_type(&self) -> LockType;
```

Comment:
Would it be okay to document that the lock type is related to artifacts?

Reply:
Oh, I see it's documented in `LockType`, so probably no need to do it here too.

Reply:
Added them anyway, maybe helps when looking at the `_cmd.rs` files 👍